src/systems/optical_tactile_plugin/OpticalTactilePlugin.cc

```diff
@@ -0,0 +1,682 @@
+#include <optional>
+#include <vector>
+
+#include <ignition/common/Profiler.hh>
+#include <ignition/plugin/Register.hh>
+#include <ignition/transport/Node.hh>
+#include <ignition/common/Time.hh>
+
+#include <sdf/Element.hh>
+
+#include "ignition/gazebo/components/ContactSensor.hh"
+#include "ignition/gazebo/components/ContactSensorData.hh"
+#include "ignition/gazebo/components/Collision.hh"
+#include "ignition/gazebo/components/DepthCamera.hh"
+#include "ignition/gazebo/components/Link.hh"
+#include "ignition/gazebo/components/Name.hh"
+#include "ignition/gazebo/components/Sensor.hh"
+#include "ignition/gazebo/components/Pose.hh"
+
+#include "ignition/gazebo/Model.hh"
+#include "ignition/gazebo/Util.hh"
+
+#include "OpticalTactilePlugin.hh"
+
+using namespace ignition;
+using namespace gazebo;
+using namespace systems;
+
+class ignition::gazebo::systems::OpticalTactilePluginPrivate
+{
+  /// \brief Load the Contact sensor from an sdf element
+  /// \param[in] _sdf SDF element describing the Contact sensor
+  /// \param[in] _ecm Immutable reference to the EntityComponentManager
+  public: void Load(const EntityComponentManager &_ecm,
+                    const sdf::ElementPtr &_sdf);
+
+  /// \brief Actual function that enables the plugin.
+  /// \param[in] _value True to enable plugin.
+  public: void Enable(const bool _value);
+
+  /// \brief Filters out new collisions fetched not related to the sensors
+  /// \param[in] _ecm Immutable reference to the EntityComponentManager
+  /// \param[in] _entities Collision entities to filter
+  public: void FilterOutCollisions(const EntityComponentManager &_ecm,
+                                   const std::vector<Entity> &_entities);
+
+  /// \brief Interpolates contact data
+  /// \param[in] _contactMsg Message containing the data to interpolate
+  public: void InterpolateData(const ignition::msgs::Contact &_contactMsg);
+
+  /// \brief Callback for the Depth Camera
+  /// \param[in] _msg Message from the subscribed topic
+  public: void DepthCameraCallback(
+               const ignition::msgs::PointCloudPacked &_msg);
+
+  /// \brief Unpacks the point cloud retrieved by the Depth Camera
+  /// into XYZ data
+  /// \param[in] _msg Message containing the data to unpack
+  public: void UnpackPointCloudMsg(
+               const ignition::msgs::PointCloudPacked &_msg);
+
+  /// \brief Computes the normal forces of the Optical Tactile sensor
+  /// \param[in] _msg Message returned by the Depth Camera
+  /// \param[in] _visualizeForces Whether to visualize the forces or not
+  ///
+  /// Implementation inspired by
+  /// https://stackoverflow.com/questions/
+  /// 34644101/calculate-surface-normals-from-depth-image-
+  /// using-neighboring-pixels-cross-produc
+  public: void ComputeNormalForces(
+               const ignition::msgs::PointCloudPacked &_msg,
+               const bool _visualizeForces);
+
+  /// \brief Resolution of the sensor in pixels to skip.
+  public: int resolution{100};
```

Comment:
Fair point! However, I'll keep the `int` type and check for positive values. If the user specifies a negative value in the `.sdf` file, `_sdf->Get<uint64_t>("resolution")` would convert it to some really big positive number. Besides, seems that [Google Style Guide](https://google.github.io/styleguide/cppguide.html#Integer_Types) doesn't recommend using unsigned types, so that's another good reason.